src/Pos/Block/Worker.hs

```diff
@@ -115,18 +115,20 @@ blkOnNewSlotImpl (slotId@SlotId {..}) sendActions = do
                       map (bprint pairF) (zip [0 :: Int ..] $ toList leaders)
         logLeadersF $ sformat ("Current slot leader: "%build) leader
         logDebugS $ sformat ("Available to use lightweight PSKs: "%listJson) validCerts
-        heavyPskM <- getPSKByIssuerAddressHash leader
-        logDebug $ "Does someone have cert for this slot: " <> show (isJust heavyPskM)
-        let heavyWeAreDelegate = maybe False ((== ourPk) . pskDelegatePk) heavyPskM
-        let heavyWeAreIssuer = maybe False ((== ourPk) . pskIssuerPk) heavyPskM
+        heavyPskIssuerM <- getPskByIssuer (Right leader)
+        let heavyWeAreIssuer = maybe False ((== ourPk) . pskIssuerPk) heavyPskIssuerM
```

Comment:
Also we shouldn't create a block if we lightly delegated to somebody, because it will lead to a fork. We can't enforce it by protocol, but we should enforce it by implementation.